cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/SettableArguments.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.api.action;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Provides access to the pipeline arguments which can be updated.
+ */
+public interface SettableArguments extends Iterable<Map.Entry<String, String>> {
```

Comment:
Currently only `SettableArguments` is used (by `Action` plugins). `Arguments` can be added later if required. Since this interface exposes `set` method calling the class as `SettableArguments` similar to the `SettableFuture`. However no strong opinion here, please let me know if name should be changed to `MutableArgument`.